metric/doc.go

```diff
@@ -26,9 +26,47 @@ instruments are created by a [Meter] which itself is created by a
 as a starting point when instrumenting. This can be done directly, or by using
 the OpenTelemetry global MeterProvider via [GetMeterProvider]. Using an
 appropriately named [Meter] from the accepted [MeterProvider], instrumentation
-can then be built from the [Meter]'s instruments. See
-[go.opentelemetry.io/otel/metric/instrument] for documentation on each
-instrument and its intended use.
+can then be built from the [Meter]'s instruments.
+
+# Instruments
+
+Each instrument is designed to make measurements of a particular type. Broadly,
+all instruments fall into two overlapping logical categories: asynchronous or
+synchronous, and int64 or float64.
+
+All synchronous instruments ([Int64Counter], [Int64UpDownCounter],
+[Int64Histogram], [Float64Counter], [Float64UpDownCounter], [Float64Histogram])
+are used to measure the operation and performance of source code during the
+source code execution. These instruments only make measurements when the source
+code they instrument is run.
+
+All asynchronous instruments ([Int64ObservableCounter],
+[Int64ObservableUpDownCounter], [Int64ObservableGauge],
+[Float64ObservableCounter], [Float64ObservableUpDownCounter],
+[Float64ObservableGauge]) are used to measure metrics outside of the execution
+of source code. They are said to make "observations" via a callback function
+called once every measurement collection cycle.
```

Comment:
How to add "observations" for histograms? Is it worth calling out that it is not possible?

Reply:
This is a copy/paste from the other package. I'd rather not edit this in this PR.

Feel free to add a PR once this merges.